management/virtualmachine/entities.go

```diff
@@ -11,6 +11,16 @@ type VirtualMachineClient struct {
 	client management.Client
 }
 
+type VMCreateConfiguration struct {
+	DnsName            string
+	InstanceSize       string
+	ImageName          string
+	Location           string
+	StorageAccountType string
+	role               *Role
+	client             VirtualMachineClient
```

Comment:
`client` definitely should not be here. having this here means a design mistake.